bin/aws-configure

```diff
@@ -39,72 +31,55 @@ To add a new profile, you MUST specify following environment variables:
 
 You can optionally specify '<PROFILE>_AWS_DEFAULT_REGION' to determine the default region for this profile.
 
-If no PROFILE is specified, ${0##*/} will defer to the value specified in AWS_PROFILE.
-
 SUBCOMMANDS:
     is-configured PROFILE     Returns '0' if the profile is configured. Otherwise, it returns '1'.
 DOC
 }
 
 function setup_profile() {
-  profile="$1"
-  profile_env="$(echo "${profile^^}" | tr '-' '_')"
+  profile=$1
+  profile_env="$(echo "$profile" | awk '{ print toupper($0) }' | tr '-' '_')"
 
-  enforce_requirements
-
-  if [[ -z "$profile" ]];
-  then
-    echo -e "${0##*/}: No profile specified. You must specify an AWS profile to configure.\n"
+  if [[ -z "$profile" ]]; then
+    echo -e "${0##*/}: No profile specified. You must specify an AWS profile to configure.\\n"
     usage
     exit 2
   fi
 
-  [[ ! -d $HOME/.aws ]] && mkdir -p $HOME/.aws > /dev/null
+  [[ ! -d "$HOME/.aws" ]] && mkdir -p "$HOME/.aws" > /dev/null
```

Comment:
This can be just:
```bash
mkdir -p "$HOME/.aws"
```
There won't be any output whether or not the directory existed beforehand, so no need to redirect.